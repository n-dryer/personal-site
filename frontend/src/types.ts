import { LucideIcon } from 'lucide-react';

export interface Project {
  name: string;
  description: string;
}

export interface Experience {
  id: string;
  title: string;
  company: string;
  location: string;
  date: string;
  description: string;
  icon: LucideIcon;
  achievements: string[];
  technologies: string[];
  projects: Project[];
}

export interface Skill {
  id: string;
  name: string;
  level: number;
  icon?: string;
}

export interface SocialLink {
  name: string;
  url: string;
}

export interface UserData {
  fullName: string;
  bioLine: string;
  photoUrl: string;
  email: string;
  phone: string;
  location: string;
  socialLinks: SocialLink[];
  resumeUrl: string;
}

// Legacy interfaces to maintain compatibility during migration
export interface LegacyExperience {
  title: string;
  company: string;
  location: string;
  dates: string;
  blurb: string[];
}
