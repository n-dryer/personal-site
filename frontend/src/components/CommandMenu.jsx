import React, { useState, useEffect, useRef } from 'react';
import { Command } from 'cmdk';
import { Download, Mail, Linkedin, Github, Moon, Sun, ChevronRight } from 'lucide-react';

const CommandMenu = ({ userData, toggleTheme, darkMode }) => {
  const [open, setOpen] = useState(false);
  const [inputValue, setInputValue] = useState('');
  const inputRef = useRef(null);
  const listRef = useRef(null);

  // Define command items
  const items = [
    { 
      label: "Download résumé", 
      icon: Download, 
      href: "/nathan_dryer_resume.pdf", 
      shortcut: "⏎", 
      primary: true 
    },
    { 
      label: "Email Nate", 
      icon: Mail, 
      href: `mailto:${userData.email}` 
    },
    { 
      label: "LinkedIn", 
      icon: Linkedin, 
      href: userData.socialLinks.find(link => link.name === "LinkedIn")?.url || "#" 
    },
    { 
      label: "GitHub", 
      icon: Github, 
      href: userData.socialLinks.find(link => link.name === "GitHub")?.url || "#" 
    },
    { 
      label: `Toggle ${darkMode ? 'light' : 'dark'} mode`, 
      icon: darkMode ? Sun : Moon, 
      action: "theme" 
    },
  ];

  // Filter items based on search input
  const filteredItems = inputValue === '' 
    ? items 
    : items.filter(item => 
        item.label.toLowerCase().includes(inputValue.toLowerCase())
      );

  // Handle keyboard shortcuts
  useEffect(() => {
    const handleKeyDown = (e) => {
      // Open command palette with Cmd+K or Ctrl+K
      if ((e.metaKey || e.ctrlKey) && e.key === 'k') {
        e.preventDefault();
        setOpen(true);
      }
    };

    window.addEventListener('keydown', handleKeyDown);
    return () => window.removeEventListener('keydown', handleKeyDown);
  }, []);

  // Auto-focus input on palette open only on desktop
  useEffect(() => {
    if (open && window.innerWidth >= 768) {
      setTimeout(() => {
        inputRef.current?.focus();
      }, 100);
    }
  }, [open]);

  // Handle email copy
  const handleEmailCopy = () => {
    navigator.clipboard.writeText(userData.email);
    // You could add a toast notification here
    setOpen(false);
  };

  return (
    <Command
      label="Site command palette"
      className="w-full max-w-full md:max-w-md mx-auto rounded-2xl bg-white/65 dark:bg-slate-900/50
                backdrop-blur-sm shadow-lg ring-1 ring-black/5 dark:ring-white/10
                focus-within:ring-blue-500 dark:focus-within:ring-blue-400 transition"
    >
      <Command.Input
        ref={inputRef}
        value={inputValue}
        onValueChange={setInputValue}
        placeholder="Type a command…"
        className="h-12 w-full bg-transparent px-4 text-base text-gray-800 dark:text-white outline-none"
        autoComplete="off"
      />
      <Command.List 
        ref={listRef}
        className="grid gap-1 p-2"
      >
        {filteredItems.map(({ label, icon: Icon, href, shortcut, primary, action }) => (
          <Command.Item 
            key={label} 
            className="rounded-xl focus:bg-slate-100 dark:focus:bg-white/5 outline-none"
            role="menuitem"
          >
            {href ? (
              <a
                href={href}
                className={`flex items-center gap-3 px-4 py-3 rounded-xl w-full min-h-[48px]
                          ${primary 
                            ? "bg-gradient-to-r from-blue-500 to-blue-600 text-white hover:from-blue-600 hover:to-blue-700" 
                            : "hover:bg-slate-100 dark:hover:bg-white/5"}`}
                download={href.includes('resume') ? "Nathan_Dryer_Resume.pdf" : undefined}
              >
                <div className={`flex items-center justify-center w-8 h-8 ${primary 
                  ? "bg-white/20 rounded-full" 
                  : "text-blue-500 dark:text-blue-400"}`}>
                  <Icon className="w-5 h-5 shrink-0" />
                </div>
                <span className="grow text-left">{label}</span>
                {shortcut && (
                  <kbd className="px-2 py-1 text-xs bg-black/5 dark:bg-white/10 rounded">
                    {shortcut}
                  </kbd>
                )}
                {!primary && !shortcut && <ChevronRight className="w-4 h-4 opacity-50" />}
              </a>
            ) : (
              <button
                onClick={() => {
                  if (action === "theme") {
                    toggleTheme();
                  }
                  setOpen(false);
                }}
                className="flex items-center gap-3 px-4 py-3 rounded-xl w-full min-h-[48px]
                          hover:bg-slate-100 dark:hover:bg-white/5"
              >
                <div className="flex items-center justify-center w-8 h-8 text-blue-500 dark:text-blue-400">
                  <Icon className="w-5 h-5 shrink-0" />
                </div>
                <span className="grow text-left text-gray-800 dark:text-white">{label}</span>
              </button>
            )}
          </Command.Item>
        ))}
      </Command.List>
    </Command>
  );
};

export default CommandMenu;
