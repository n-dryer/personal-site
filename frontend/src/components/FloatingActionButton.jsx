import React, { useState, useEffect, useRef } from 'react';
import { motion, AnimatePresence } from 'framer-motion';
import CopyToClipboard from 'react-copy-to-clipboard';

const FloatingActionButton = ({ email }) => {
  const [isOpen, setIsOpen] = useState(false);
  const [showToast, setShowToast] = useState(false);
  const [toastMessage, setToastMessage] = useState('');
  const containerRef = useRef(null);

  useEffect(() => {
    const handleClickOutside = (e) => {
      if (containerRef.current && !containerRef.current.contains(e.target)) {
        setIsOpen(false);
      }
    };

    if (isOpen) {
      document.addEventListener('mousedown', handleClickOutside);
    }
    
    return () => {
      document.removeEventListener('mousedown', handleClickOutside);
    };
  }, [isOpen]);

  const handleCopy = () => {
    setToastMessage('Email copied to clipboard!');
    setShowToast(true);
    setTimeout(() => setShowToast(false), 2000);
  };

  const scrollToTop = () => {
    document.getElementById('top')?.scrollIntoView({ behavior: 'smooth' });
  };

  const scrollToTimeline = () => {
    document.getElementById('timeline')?.scrollIntoView({ behavior: 'smooth' });
  };

  const scrollToSkills = () => {
    document.getElementById('skills')?.scrollIntoView({ behavior: 'smooth' });
  };

  const scrollToBottom = () => {
    document.getElementById('end')?.scrollIntoView({ behavior: 'smooth' });
  };

  const downloadResume = () => {
    const link = document.createElement('a');
    link.href = '/nathan_dryer_resume.pdf';
    link.download = 'Nathan_Dryer_Resume.pdf';
    link.click();
    
    setToastMessage('Resume downloaded!');
    setShowToast(true);
    setTimeout(() => setShowToast(false), 2000);
  };

  // Button configuration
  const buttons = [
    { label: "Top", icon: "arrow-up", onClick: scrollToTop },
    { label: "Timeline", icon: "timeline", onClick: scrollToTimeline },
    { label: "Skills", icon: "lightbulb", onClick: scrollToSkills },
    { label: "Contact", icon: "chevron-down", onClick: scrollToBottom },
    { 
      label: "Email", 
      icon: "mail",
      onClick: handleCopy,
      copyToClipboard: true,
      text: email
    },
    { 
      label: "Resume", 
      icon: "document-download",
      onClick: downloadResume 
    }
  ];

  // Circular menu layout calculations
  const getButtonPosition = (index, total, radius = 80) => {
    // Start from the top position and distribute buttons evenly in a circle
    const angleOffset = -90; // Start from top (negative y-axis in screen coordinates)
    const angleStep = (360 / total);
    const angle = ((index * angleStep) + angleOffset) * (Math.PI / 180);
    
    return {
      x: Math.cos(angle) * radius,
      y: Math.sin(angle) * radius
    };
  };

  // SVG path map
  const iconPathMap = {
    "arrow-up": "M5 15l7-7 7 7",
    "timeline": "M4 5h16M4 12h16M4 19h7",
    "lightbulb": "M9.663 17h4.673M12 3v1m6.364 1.636l-.707.707M21 12h-1M4 12H3m3.343-5.657l-.707-.707m2.828 9.9a5 5 0 117.072 0l-.548.547A3.374 3.374 0 0014 18.469V19a2 2 0 11-4 0v-.531c0-.895-.356-1.754-.988-2.386l-.548-.547z",
    "chevron-down": "M19 9l-7 7-7-7",
    "mail": "M3 8l7.89 5.26a2 2 0 002.22 0L21 8M5 19h14a2 2 0 002-2V7a2 2 0 00-2-2H5a2 2 0 00-2 2v10a2 2 0 002 2z",
    "document-download": "M12 10v6m0 0l-3-3m3 3l3-3m2 8H7a2 2 0 01-2-2V5a2 2 0 012-2h5.586a1 1 0 01.707.293l5.414 5.414a1 1 0 01.293.707V19a2 2 0 01-2 2z"
  };

  return (
    <>
      <div ref={containerRef} className="fixed bottom-8 right-8 z-50">
        <div className="relative">
          <AnimatePresence>
            {isOpen && (
              <div className="absolute top-0 left-0">
                {buttons.map((button, index) => {
                  const position = getButtonPosition(index, buttons.length);
                  
                  if (button.copyToClipboard) {
                    return (
                      <CopyToClipboard key={index} text={button.text} onCopy={button.onClick}>
                        <motion.button
                          className="menu-button"
                          initial={{ scale: 0, x: 0, y: 0, opacity: 0 }}
                          animate={{ 
                            scale: 1, 
                            x: position.x, 
                            y: position.y,
                            opacity: 1,
                          }}
                          exit={{ 
                            scale: 0,
                            x: 0,
                            y: 0,
                            opacity: 0,
                            transition: { duration: 0.2 }
                          }}
                          transition={{ 
                            type: "spring",
                            stiffness: 350,
                            damping: 25,
                            delay: index * 0.04,
                          }}
                          aria-label={button.label}
                          whileHover={{ 
                            scale: 1.1, 
                            backgroundColor: "#4F46E5",
                            color: "white"
                          }}
                          whileTap={{ scale: 0.95 }}
                        >
                          <svg xmlns="http://www.w3.org/2000/svg" className="h-5 w-5" fill="none" viewBox="0 0 24 24" stroke="currentColor">
                            <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d={iconPathMap[button.icon]} />
                          </svg>
                        </motion.button>
                      </CopyToClipboard>
                    );
                  }

                  return (
                    <motion.button
                      key={index}
                      className="menu-button"
                      initial={{ scale: 0, x: 0, y: 0, opacity: 0 }}
                      animate={{ 
                        scale: 1, 
                        x: position.x, 
                        y: position.y,
                        opacity: 1,
                      }}
                      exit={{ 
                        scale: 0,
                        x: 0,
                        y: 0,
                        opacity: 0,
                        transition: { duration: 0.2 }
                      }}
                      transition={{ 
                        type: "spring",
                        stiffness: 350,
                        damping: 25,
                        delay: index * 0.04,
                      }}
                      onClick={button.onClick}
                      aria-label={button.label}
                      whileHover={{ 
                        scale: 1.1, 
                        backgroundColor: "#4F46E5",
                        color: "white"
                      }}
                      whileTap={{ scale: 0.95 }}
                    >
                      <svg xmlns="http://www.w3.org/2000/svg" className="h-5 w-5" fill="none" viewBox="0 0 24 24" stroke="currentColor">
                        <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d={iconPathMap[button.icon]} />
                      </svg>
                    </motion.button>
                  );
                })}
              </div>
            )}
          </AnimatePresence>

          <motion.button
            className="main-button"
            onClick={() => setIsOpen(!isOpen)}
            whileHover={{ scale: 1.1 }}
            whileTap={{ scale: 0.95 }}
            animate={isOpen ? { rotate: 135 } : { rotate: 0 }}
            transition={{ type: "spring", stiffness: 500, damping: 30 }}
            aria-label={isOpen ? "Close menu" : "Open menu"}
            aria-expanded={isOpen}
          >
            <svg xmlns="http://www.w3.org/2000/svg" className="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor">
              <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M12 6v6m0 0v6m0-6h6m-6 0H6" />
            </svg>
          </motion.button>
        </div>
      </div>

      <AnimatePresence>
        {showToast && (
          <motion.div
            className="toast"
            initial={{ opacity: 0, y: 50 }}
            animate={{ opacity: 1, y: 0 }}
            exit={{ opacity: 0 }}
            transition={{ duration: 0.3 }}
            role="alert"
          >
            {toastMessage}
          </motion.div>
        )}
      </AnimatePresence>
    </>
  );
};

export default FloatingActionButton;
