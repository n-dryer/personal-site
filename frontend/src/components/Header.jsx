import React from 'react';
import { motion } from 'framer-motion';
import { ParallaxBanner } from 'react-scroll-parallax';

const Header = ({ userData }) => {
  return (
    <ParallaxBanner
      layers={[
        { speed: -20, expanded: false, children: (
          <div className="absolute inset-0 bg-gradient-to-br from-teal-400 to-teal-600 opacity-100">
            {/* Hexagon Pattern - created with multiple divs */}
            <div className="hexagon-pattern">
              {Array.from({ length: 20 }).map((_, i) => (
                <div key={i} className="hexagon"></div>
              ))}
            </div>
          </div>
        )},
      ]}
      className="relative h-screen flex items-center justify-center overflow-hidden"
      id="top"
    >
      <div className="relative z-10 container mx-auto px-6 py-8">
        <div className="flex flex-col items-center justify-center h-full text-center">
          <motion.div 
            className="mb-10"
            initial={{ opacity: 0, y: 20 }}
            animate={{ opacity: 1, y: 0 }}
            transition={{ duration: 0.6 }}
          >
            <div className="profile-circle">
              <img 
                src={userData.photoUrl} 
                alt={userData.fullName}
                className="w-full h-full object-cover"
              />
            </div>
          </motion.div>
          
          <motion.div 
            className="text-center text-white max-w-xl"
            initial={{ opacity: 0 }}
            animate={{ opacity: 1 }}
            transition={{ duration: 0.7, delay: 0.2 }}
          >
            <motion.h1 
              className="text-5xl md:text-6xl font-bold mb-4 font-sans"
              initial={{ opacity: 0, y: 20 }}
              animate={{ opacity: 1, y: 0 }}
              transition={{ duration: 0.5, delay: 0.3 }}
            >
              {userData.fullName}
            </motion.h1>
            
            <motion.div
              className="flex flex-col items-center space-y-6"
              initial={{ opacity: 0, y: 20 }}
              animate={{ opacity: 1, y: 0 }}
              transition={{ duration: 0.5, delay: 0.4 }}
            >
              <p className="text-xl text-white font-light">
                {userData.bioLine}
              </p>
              
              <motion.a
                href="/nathan_dryer_resume.pdf"
                download="Nathan_Dryer_Resume.pdf"
                className="modern-button"
                whileHover={{ scale: 1.05 }}
                whileTap={{ scale: 0.95 }}
              >
                <svg className="w-4 h-4 mr-2" fill="none" viewBox="0 0 24 24" stroke="currentColor">
                  <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M12 10v6m0 0l-3-3m3 3l3-3m2 8H7a2 2 0 01-2-2V5a2 2 0 012-2h5.586a1 1 0 01.707.293l5.414 5.414a1 1 0 01.293.707V19a2 2 0 01-2 2z" />
                </svg>
                Download Resume
              </motion.a>
            </motion.div>
          </motion.div>
        </div>
      </div>
      
      <div className="absolute bottom-16 left-1/2 transform -translate-x-1/2">
        <motion.div 
          className="scroll-down-circle"
          initial={{ opacity: 0, y: -20 }}
          animate={{ opacity: 1, y: 0 }}
          transition={{ delay: 1, duration: 0.5, repeat: Infinity, repeatType: "reverse" }}
          onClick={() => document.getElementById('timeline')?.scrollIntoView({ behavior: 'smooth' })}
        >
          <svg 
            xmlns="http://www.w3.org/2000/svg" 
            className="h-6 w-6 text-white" 
            fill="none" 
            viewBox="0 0 24 24" 
            stroke="currentColor"
            aria-label="Scroll down to timeline"
            role="button"
            tabIndex={0}
          >
            <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M19 14l-7 7m0 0l-7-7m7 7V3" />
          </svg>
        </motion.div>
      </div>
    </ParallaxBanner>
  );
};

export default Header;
