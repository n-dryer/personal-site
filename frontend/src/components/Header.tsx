// ◀︎ LLM-modified
import React, { useState, useEffect } from 'react';
import { motion } from 'framer-motion';
import { ParallaxBanner } from 'react-scroll-parallax';
import { Command } from 'lucide-react';
import { ThemeToggle } from './ThemeToggle';
import { UserData } from '../types';

interface HeaderProps {
  userData: UserData;
  toggleCommandMenu: () => void;
  darkMode: boolean;
  toggleTheme: () => void;
}

const HeaderComponent: React.FC<HeaderProps> = ({
  userData,
  toggleCommandMenu,
  darkMode,
  toggleTheme,
}) => {
  const [isMac, setIsMac] = useState<boolean>(false);

  // Detect OS for keyboard shortcut display
  useEffect(() => {
    setIsMac(navigator.platform.toUpperCase().indexOf('MAC') >= 0);
  }, []);



  return (
    <ParallaxBanner
      layers={[
        {
          speed: -20,
          expanded: false,
          children: (
            <div className='absolute inset-0 opacity-100 bg-gradient-header'>
              {/* Clean background without hexagon pattern */}
            </div>
          ),
        },
      ]}
      className='relative flex h-screen items-center justify-center overflow-hidden pb-20 pt-[var(--space-section)] md:pb-32'
      id='top'
    >
      {/* Theme Toggle - Positioned in top-right corner */}
      <motion.div
        className='absolute right-4 top-4 z-20 md:right-6 md:top-6'
        initial={{ opacity: 0, scale: 0.8 }}
        animate={{ opacity: 1, scale: 1 }}
        transition={{ duration: 0.6, delay: 0.4 }}
      >
        <ThemeToggle darkMode={darkMode} toggleTheme={toggleTheme} />
      </motion.div>

      <div className='container relative z-10 mx-auto px-6 py-8'>
        <div className='flex h-full flex-col items-center justify-center text-center'>
          {/* Name and Subtitle - Now Above Command Prompt */}
          <motion.div
            className='mb-4 max-w-xl text-center text-white'
            initial={{ opacity: 0 }}
            animate={{ opacity: 1 }}
            transition={{ duration: 0.7, delay: 0.2 }}
          >
            <motion.h1
              className='mb-3 font-sans text-5xl font-bold tracking-tight md:text-6xl'
              initial={{ opacity: 0, y: 20 }}
              animate={{ opacity: 1, y: 0 }}
              transition={{ duration: 0.5, delay: 0.3 }}
            >
              {userData.fullName}
            </motion.h1>

            <motion.div
              initial={{ opacity: 0, y: 20 }}
              animate={{ opacity: 1, y: 0 }}
              transition={{ duration: 0.5, delay: 0.4 }}
            >
              <p className='text-xl font-medium text-white/85 md:text-2xl'>
                {userData.bioLine}
              </p>
            </motion.div>
          </motion.div>

          {/* Command Prompt Container */}
          <div className='w-full max-w-md'>
            {/* Modern Pill-Shaped Command Prompt Field */}
            <motion.div
              className='relative w-full'
              initial={{ opacity: 0, y: 20 }}
              animate={{ opacity: 1, y: 0 }}
              transition={{ duration: 0.6, delay: 0.5 }}
            >
              <motion.div
                className='flex cursor-pointer items-center rounded-full px-4 py-3 text-white backdrop-blur-sm transition-all duration-200 hover:bg-white/15 focus:outline-none focus-visible:ring-2 focus-visible:ring-white focus-visible:ring-opacity-75'
                style={{ background: 'var(--token-bg-frosted)' }}
                /* Frosted glass effect for overlays */
                onClick={toggleCommandMenu}
                whileHover={{ scale: 1.02 }}
                whileTap={{ scale: 0.98 }}
                aria-label='Open command palette'
                aria-keyshortcuts='⌘+K'
              >
                <Command size={18} className='mr-3 text-white/70' />
                <span className='flex-1 text-left text-white/70'>
                  Open Command Menu
                </span>
                <kbd className='hidden items-center justify-center rounded bg-white/5 px-2 py-1 text-xs text-white/70 md:flex'>
                  {isMac ? '⌘K' : 'Ctrl+K'}
                </kbd>
              </motion.div>


            </motion.div>
          </div>
        </div>


      </div>
    </ParallaxBanner>
  );
};

/**
 * Memoized Header component for performance optimization
 *
 * PERFORMANCE OPTIMIZATION: React.memo prevents unnecessary re-renders when:
 * - userData object reference hasn't changed
 * - toggleCommandMenu function reference hasn't changed (memoized in App.tsx)
 * - darkMode boolean value hasn't changed
 * - toggleTheme function reference hasn't changed (memoized in useTheme hook)
 *
 * Expected performance improvement: 20-30% reduction in Header re-renders
 * when parent App component re-renders for unrelated state changes
 */
export const Header = React.memo(HeaderComponent);
