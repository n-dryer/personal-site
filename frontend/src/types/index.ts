// Re-export from portfolio.ts for backward compatibility
export * from './portfolio';
