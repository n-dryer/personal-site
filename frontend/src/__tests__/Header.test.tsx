import React from 'react';
import { render } from '@testing-library/react';
import { Header } from '../components/Header';
import { UserData } from '../types';

// Mock external dependencies
jest.mock('framer-motion', () => ({
  motion: {
    div: ({ children, ...props }: React.PropsWithChildren<any>) => (
      <div data-testid='motion-div' {...props}>
        {children}
      </div>
    ),
    h1: ({ children, ...props }: React.PropsWithChildren<any>) => (
      <h1 data-testid='motion-h1' {...props}>
        {children}
      </h1>
    ),
    button: ({ children, ...props }: React.PropsWithChildren<any>) => (
      <button data-testid='motion-button' {...props}>
        {children}
      </button>
    ),
  },
  AnimatePresence: ({ children }: React.PropsWithChildren<any>) => (
    <>{children}</>
  ),
}));

jest.mock('react-scroll-parallax', () => ({
  ParallaxBanner: ({ children, ...props }: React.PropsWithChildren<any>) => (
    <div data-testid='parallax-banner' {...props}>
      {children}
    </div>
  ),
}));

jest.mock('lucide-react', () => ({
  Command: () => <div data-testid='command-icon'>Command Icon</div>,
  Sun: () => <div data-testid='sun-icon'>Sun Icon</div>,
  Moon: () => <div data-testid='moon-icon'>Moon Icon</div>,
}));



describe('Header Component', () => {
  // Mock props
  const mockUserData: UserData = {
    fullName: 'Test User',
    bioLine: 'Test Bio',
    photoUrl: '/test.jpg',
    email: 'test@example.com',
    phone: '(123) 456-7890',
    location: 'Test Location',
    socialLinks: [
      { name: 'GitHub', url: 'https://github.com/test' },
      { name: 'LinkedIn', url: 'https://linkedin.com/in/test' },
      { name: 'Personal Site', url: 'https://test.com' },
    ],
    resumeUrl: '/test_resume.pdf',
  };

  const mockToggleTheme = jest.fn();
  const mockToggleCommandMenu = jest.fn();

  it('renders correctly', () => {
    const { container } = render(
      <Header
        userData={mockUserData}
        toggleCommandMenu={mockToggleCommandMenu}
        darkMode={false}
        toggleTheme={mockToggleTheme}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('renders with command menu functionality', () => {
    const { container } = render(
      <Header
        userData={mockUserData}
        toggleCommandMenu={mockToggleCommandMenu}
        darkMode={false}
        toggleTheme={mockToggleTheme}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
