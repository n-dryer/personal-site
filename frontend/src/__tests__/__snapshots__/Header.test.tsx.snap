// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders correctly 1`] = `
<div>
  <div
    class="relative h-screen flex items-center justify-center overflow-hidden pb-20 md:pb-32"
    data-testid="parallax-banner"
    id="top"
    layers="[object Object]"
  >
    <button
      aria-label="Switch to dark mode"
      class="absolute top-6 right-6 z-20 p-2 rounded-full bg-white/10 backdrop-blur-sm text-white hover:bg-white/20 transition-colors focus:outline-none focus-visible:ring-2 focus-visible:ring-white focus-visible:ring-opacity-75"
      data-testid="motion-button"
      whilehover="[object Object]"
      whiletap="[object Object]"
    >
      <div
        data-testid="moon-icon"
      >
        Moon Icon
      </div>
    </button>
    <div
      class="relative z-10 container mx-auto px-6 py-8"
    >
      <div
        class="flex flex-col items-center justify-center h-full text-center"
      >
        <div
          animate="[object Object]"
          class="text-center text-white max-w-xl mb-8"
          data-testid="motion-div"
          initial="[object Object]"
          transition="[object Object]"
        >
          <h1
            animate="[object Object]"
            class="text-5xl md:text-6xl font-bold mb-6 font-sans tracking-tight"
            data-testid="motion-h1"
            initial="[object Object]"
            transition="[object Object]"
          >
            Test User
          </h1>
          <div
            animate="[object Object]"
            data-testid="motion-div"
            initial="[object Object]"
            transition="[object Object]"
          >
            <p
              class="text-2xl md:text-3xl text-blue-300 font-medium"
            >
              Test Bio
            </p>
          </div>
        </div>
        <div
          animate="[object Object]"
          class="relative w-full max-w-md"
          data-testid="motion-div"
          initial="[object Object]"
          transition="[object Object]"
        >
          <div
            aria-keyshortcuts="⌘+K"
            aria-label="Open command palette"
            class="flex items-center px-4 py-3 rounded-full bg-white/10 backdrop-blur-sm border border-white/20 text-white cursor-pointer hover:bg-white/15 transition-all duration-200 focus:outline-none focus-visible:ring-2 focus-visible:ring-white focus-visible:ring-opacity-75"
            data-testid="motion-div"
            whilehover="[object Object]"
            whiletap="[object Object]"
          >
            <div
              data-testid="command-icon"
            >
              Command Icon
            </div>
            <span
              class="text-white/70 flex-1 text-left"
            >
              Search or type a command...
            </span>
            <kbd
              class="hidden md:flex items-center justify-center rounded border border-white/20 bg-white/5 px-2 py-1 text-xs text-white/70"
            >
              Ctrl+K
            </kbd>
          </div>
        </div>
      </div>
    </div>
    <div
      animate="[object Object]"
      class="absolute -bottom-32 left-1/2 transform -translate-x-1/2"
      data-testid="motion-div"
      initial="[object Object]"
      transition="[object Object]"
    >
      <div
        animate="bounce"
        class="relative"
        data-testid="motion-div"
        variants="[object Object]"
      >
        <div
          class="absolute inset-0 rounded-full bg-blue-500/20 blur-md"
        />
        <div
          class="relative z-10"
        >
          <div
            data-testid="scroll-cue"
          >
            Scroll Cue
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Header Component renders with dark mode 1`] = `
<div>
  <div
    class="relative h-screen flex items-center justify-center overflow-hidden pb-20 md:pb-32"
    data-testid="parallax-banner"
    id="top"
    layers="[object Object]"
  >
    <button
      aria-label="Switch to light mode"
      class="absolute top-6 right-6 z-20 p-2 rounded-full bg-white/10 backdrop-blur-sm text-white hover:bg-white/20 transition-colors focus:outline-none focus-visible:ring-2 focus-visible:ring-white focus-visible:ring-opacity-75"
      data-testid="motion-button"
      whilehover="[object Object]"
      whiletap="[object Object]"
    >
      <div
        data-testid="sun-icon"
      >
        Sun Icon
      </div>
    </button>
    <div
      class="relative z-10 container mx-auto px-6 py-8"
    >
      <div
        class="flex flex-col items-center justify-center h-full text-center"
      >
        <div
          animate="[object Object]"
          class="text-center text-white max-w-xl mb-8"
          data-testid="motion-div"
          initial="[object Object]"
          transition="[object Object]"
        >
          <h1
            animate="[object Object]"
            class="text-5xl md:text-6xl font-bold mb-6 font-sans tracking-tight"
            data-testid="motion-h1"
            initial="[object Object]"
            transition="[object Object]"
          >
            Test User
          </h1>
          <div
            animate="[object Object]"
            data-testid="motion-div"
            initial="[object Object]"
            transition="[object Object]"
          >
            <p
              class="text-2xl md:text-3xl text-blue-300 font-medium"
            >
              Test Bio
            </p>
          </div>
        </div>
        <div
          animate="[object Object]"
          class="relative w-full max-w-md"
          data-testid="motion-div"
          initial="[object Object]"
          transition="[object Object]"
        >
          <div
            aria-keyshortcuts="⌘+K"
            aria-label="Open command palette"
            class="flex items-center px-4 py-3 rounded-full bg-white/10 backdrop-blur-sm border border-white/20 text-white cursor-pointer hover:bg-white/15 transition-all duration-200 focus:outline-none focus-visible:ring-2 focus-visible:ring-white focus-visible:ring-opacity-75"
            data-testid="motion-div"
            whilehover="[object Object]"
            whiletap="[object Object]"
          >
            <div
              data-testid="command-icon"
            >
              Command Icon
            </div>
            <span
              class="text-white/70 flex-1 text-left"
            >
              Search or type a command...
            </span>
            <kbd
              class="hidden md:flex items-center justify-center rounded border border-white/20 bg-white/5 px-2 py-1 text-xs text-white/70"
            >
              Ctrl+K
            </kbd>
          </div>
        </div>
      </div>
    </div>
    <div
      animate="[object Object]"
      class="absolute -bottom-32 left-1/2 transform -translate-x-1/2"
      data-testid="motion-div"
      initial="[object Object]"
      transition="[object Object]"
    >
      <div
        animate="bounce"
        class="relative"
        data-testid="motion-div"
        variants="[object Object]"
      >
        <div
          class="absolute inset-0 rounded-full bg-blue-500/20 blur-md"
        />
        <div
          class="relative z-10"
        >
          <div
            data-testid="scroll-cue"
          >
            Scroll Cue
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
